### ISSUE TITLE
Idea: Ignore `vendor/` directory by default?

### ISSUE THREAD

**Issue by gylaz** [thoughtbot/hound#627]
In Rails, `vendor/` directory is usually third-party code. Should we just ignore this by default?

**Comment by teoljungberg** [comment]
I'm all for it, but there is always going to be edge-cases. How about making it configurable and ignoring it default?

**Comment by srt32** [comment]
:+1: to ignoring it by default with the ability to override the setting.

**Comment by salbertson** [comment]
What would the override look like in the config?

**Comment by teoljungberg** [comment]
@salbertson maybe something like:

```yaml
ignored_directories:
- vendor
# ...
```

**Comment by salbertson** [comment]
@teoljungberg, we can ignore `vendor` if we add this to `config/style_guides/ruby.yml`.

```
AllCops:
  Exclude:
    - 'vendor/**/*'
    - db/schema.rb
```

I was asking about what the override would look like, but I realized users can simply add `Exclude:` to their own config.

**Comment by salbertson** [comment]
@gylaz, sounds good to me. :+1:

**Comment by teoljungberg** [comment]
@salbertson I didn't know you could do that, that's handy!

Then we have covered the case of ruby, but maybe we want to support ignoring `vendor` for node and JS applications aswell?

**Comment by gylaz** [comment]
Ignoring `vendor/` via RuboCop only solves a small part of the issue -- since rarely ruby code ends up in `vendor/`. Ignoring Coffee, JS, and SCSS from `vendor/` would be much more useful.

**Comment by teoljungberg** [comment]
Closed by 396d381355340e368756459c464a0327f522f5f8
